src/components/Inputs/TextInput/TextInput.tsx

```diff
@@ -392,88 +394,120 @@ export const TextInput: FC<TextInputProps> = React.forwardRef(
-                        disabled={!allowDisabledFocus && mergedDisabled}
-                        id={mergedFormItemInput ? id : inputId}
-                        maxLength={maxlength}
-                        minLength={minlength}
-                        name={name}
-                        onChange={!allowDisabledFocus ? handleChange : null}
-                        onBlur={!allowDisabledFocus ? onBlur : null}
-                        onFocus={!allowDisabledFocus ? onFocus : null}
-                        onKeyDown={!allowDisabledFocus ? onKeyDown : null}
-                        placeholder={placeholder}
-                        required={required}
-                        role="textbox"
-                        style={style}
-                        tabIndex={0}
-                        type={numbersOnly ? 'number' : htmlType}
-                        value={inputValue}
-                        readOnly={readonly}
-                    />
-                    {iconProps && (
-                        <div className={iconClassNames}>
-                            {iconProps.path && !iconProps.imageSrc && (
-                                <Icon
-                                    {...iconProps}
-                                    path={iconProps.path}
-                                    size={inputSizeToIconSizeMap.get(
-                                        mergedSize
+                <div className={styles.expandableWrapper}>
+                    <div className={textInputGroupClassNames}>
+                        <input
+                            {...rest}
+                            ref={ref}
+                            aria-disabled={mergedDisabled}
+                            aria-label={ariaLabel}
+                            autoFocus={autoFocus}
+                            className={textInputClassNames}
+                            disabled={!allowDisabledFocus && mergedDisabled}
+                            id={mergedFormItemInput ? id : inputId}
+                            maxLength={maxlength}
+                            minLength={minlength}
+                            name={name}
+                            onChange={!allowDisabledFocus ? handleChange : null}
+                            onBlur={!allowDisabledFocus ? onBlur : null}
+                            onFocus={!allowDisabledFocus ? onFocus : null}
+                            onKeyDown={!allowDisabledFocus ? onKeyDown : null}
+                            placeholder={placeholder}
+                            required={required}
+                            role="textbox"
+                            style={style}
+                            tabIndex={0}
+                            type={numbersOnly ? 'number' : htmlType}
+                            value={inputValue}
+                            readOnly={readonly}
+                        />
+                        {expandable && iconButtonProps && (
```

Comment:
what happens when expandable and no iconButtonProps?

Reply:
The idea/assumption is that the user will pass `iconButtonProps`.
There will be not placeholder in case `iconButtonProps` is not provided.